ayaneo-platform.c

```diff
@@ -993,19 +1421,36 @@ static int ayaneo_platform_probe(struct platform_device *pdev)
                 return ret;
 
         ret = devm_device_add_group(ayaneo_led_mc.led_cdev.dev, &ayaneo_led_mc_group);
+        if (ret)
+                return ret;
+
+        if(ayaneo_ps_check_bypass_charge() >= 0) {
+            struct power_supply *psy = devm_power_supply_register_no_ws(dev, &ayaneo_ps, NULL);
+            if(psy == NULL) {
+                ret = -1;
+            }
+            ps_priv.bypass_available = 1;
+        }
         return ret;
 }
 
 static void ayaneo_platform_shutdown(struct platform_device *pdev)
 {
         kthread_stop(ayaneo_led_mc_writer_thread);
         ayaneo_led_mc_release_control();
+        if(ps_priv.bypass_available) {
+            kthread_stop(ayaneo_bypass_charge_writer_thread);
+        }
 }
 
-static void ayaneo_platform_remove(struct platform_device *pdev)
+static int ayaneo_platform_remove(struct platform_device *pdev)
```

Comment:
Did this change recently? The function signature needs to conform to the header prototype.

Reply:
I may not rember if it was just a warning or an error. But I changed it to get rid of it and the compiler went silent.

Reply:
We should base it off the latest kernel release.